lib/logstash/filters/geoip.rb

```diff
@@ -166,8 +170,12 @@ def filter(event)
       # Dont' swallow this, bubble up for unknown issue
       raise e
     end
-
-    event[@target] = geo_data_hash
+    if @merge
+      event[@target] ||= {}
+      event[@target].merge!(geo_data_hash)
```

Comment:
Because the api for accessing an event is only available as 'event.get` and `event.set`. Getting a value, then modifying that value, instead of using just 'get' and 'set' APIs is not supported and not guaranteed to work correctly.